components/profile/Tabs/General.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react'
+import { User } from '@/assets/api/auth/userSlice'
+import style from './ProfileTabs.module.scss'
+import commonStyle from '../../common/Inputs/Inputs.module.scss'
+import { FormInput, FormTextarea } from '@/components/common/Inputs/Inputs'
+import { UserProfile } from '@/assets/api/user/userTypes'
+import { useUpdateProfileMutation } from '@/assets/api/user/profileQueryApi'
+import { Loading } from '@/components/common/Loaders/Loading'
+import { serverAPI } from '@/assets/api/api'
+
+type GeneralType = {
+  userProfile: UserProfile
+}
+
+const General: React.FC<GeneralType> = ({ userProfile }) => {
+  const [updatedUserData, setUpdatedUserData] = useState<UserProfile>(userProfile)
+  const [updateProfile] = useUpdateProfileMutation()
+  const [changedFields, setChangedFields] = useState<any>({})
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleSave = async () => {
+    try {
+      setIsLoading(true)
+      // Send separate requests for each changed field
+      const promises = Object.keys(changedFields).map(async (field) => {
+        const data: any = { [field]: changedFields[field] }
+        return updateProfile(data).unwrap()
+      })
+
+      await Promise.allSettled(promises)
+      await serverAPI.profile.getProfile()
+      setIsLoading(false)
+
+      // Update the local state with the new profile data
+      // (assuming serverAPI.profile.getProfile() returns the updated user profile data)
+      const updatedProfileData: any = await serverAPI.profile.getProfile()
```

Comment:
It`s weird to use serverAPI in client, 
I felt this issue by this

Reply:
agree with serverAPI in client. needs refactoring. any ideas?

Reply:
> 

when I send 3 requests and 2 of them are ok, but 1 is failed, I as a user want to see the updated profile (with all the changed fields and with those which were not changed). that's why I need a final get-request when all mutations are done

Reply:
please write a message in the slack that you commented on issues , I accidentally went to Github and saw that you answered, otherwise it took several days